libraries/AP_Vehicle/AP_Vehicle.cpp

```diff
@@ -740,6 +749,35 @@ void AP_Vehicle::one_Hz_update(void)
     }
 }
 
+void AP_Vehicle::check_motor_noise()
+{
+#if HAL_GYROFFT_ENABLED && HAL_WITH_ESC_TELEM
+    if (!hal.util->get_soft_armed() || !gyro_fft.get_post_filter_mask() || ins.has_fft_notch()) {
+        return;
+    }
+
+    float esc_data[ESC_TELEM_MAX_ESCS];
+    const uint8_t numf = AP::esc_telem().get_motor_frequencies_hz(ESC_TELEM_MAX_ESCS, esc_data);
+    bool output_error = false;
+
+    for (uint8_t i = 0; i<numf; i++) {
+        if (is_zero(esc_data[i])) {
+            continue;
+        }
+        float energy = gyro_fft.has_noise_at_frequency_hz(esc_data[i]);
+        energy = esc_noise[i].apply(energy, 0.2f);
+        if (energy > 40.0f && AP_HAL::millis() - last_motor_noise_ms > 5000) {
+            gcs().send_text(MAV_SEVERITY_WARNING, "Significant motor noise %.1fdB on motor %u at %.1fHz", energy, i+1, esc_data[i]);
```

Comment:
better as opt-in with new FFT OPTIONS bitmask

Reply:
should also be a much more compact message

Reply:
Added to FFT_OPTIONS